routes/adminAccess.js

```diff
@@ -0,0 +1,49 @@
+var express = require('express');
+var router = express.Router();
+var userMapper = require('../mappers/userMapper');
+var isLoggedIn = require('../config/utils').isLoggedIn;
+var isAdminUser = require('../config/utils').isAdminUser;
+
+router.get('/admin',isLoggedIn, isAdminUser, function(req, res) {
+	res.render('adminAccess', { user: req.user,err: req.flash('err'),succ: req.flash('succ')});
+});
+
+router.post('/admin', isLoggedIn, isAdminUser, function(req, res) {
+	if(validEmail(req.body)){
+		userMapper.findUserByEmail(req.body.userEmail, function(error, result) {
+			if (!result) {
+				req.flash('err', 'User email not found');
+				res.redirect('/');
+			} else if (error) {
+				req.flash('err', error);
+			} else {
+
+				var adminType = 'admin';
+				var adminGrantedUser =  {name: result[0].name, email: result[0].email, password:result[0].password, type: adminType, salt:result[0].salt};
```

Comment:
I'm sure this is an issue in other parts of the codebase but just occurred to me that we should use the object returned from the DB as the base object and just reassign the field we want to change (the type in this case) rather than create a new object. This means if we ever change the user model in the future we wont have to add the new fields at each of these lines of codes.

Reply:
I'm talking about var adminGrantedUser here ^